src/raftstore/store/peer.rs

```diff
@@ -493,7 +493,16 @@ impl Peer {
                 MessageType::MsgHeartbeat => metrics.heartbeat += 1,
                 MessageType::MsgHeartbeatResponse => metrics.heartbeat_resp += 1,
                 MessageType::MsgTransferLeader => metrics.transfer_leader += 1,
-                MessageType::MsgTimeoutNow => metrics.timeout_now += 1,
+                MessageType::MsgTimeoutNow => {
```

Comment:
It's possible that the logs in range `[applied_index, self.raft_log.last_index()]` would be applied after MsgTimeoutNow is sent.

And transferee is reset after election_timeout. If the condition of `transferee.is_some() && matched_index == last_index()` is used to check for lease read. Then if a entry is applied after MsgTimeoutNow is sent, the lease_expired_time is set to a timestamp after election_timeout. When transferee is reset, the old leader will be reused to do lease read. However it's possible that a new leader is elected, and the old leader may not step down due to network partition.